p32/src/lib.rs

```diff
@@ -0,0 +1,336 @@
+use crate::TransferFundsError::{
+    ReceiverNotExistsError, SenderNotEnoughBalance, SenderNotExistsError,
+};
+
+struct User {
+    name: String,
+    credit_line: u64,
+    balance: i64,
+}
+
+impl User {
+    pub(crate) fn new(name: String, credit_line: u64, balance: i64) -> Self {
+        User {
+            name,
+            credit_line,
+            balance,
+        }
+    }
+}
+
+struct Bank {
+    users: Vec<User>,
+    name: String,
+    credit_interest: u64,
+    debit_interest: u64,
+}
+
+impl Bank {
+    pub(crate) fn merge_bank(&mut self, _other: &mut Bank) -> Bank {
+        let mut merged_users: Vec<User> = vec![];
+        // TODO: is there a function call chain to zip by a given property?
+        // Instead of:
+        // self.users.iter().zip(merged_users.iter_mut()).for_each(|(user, merged_user)| {})
+        // Try:
+        // https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.chunk_by
+        //
+        // self.users.chunk_by(|user| user.name);
+        // Not working because the lambda needs to be a predicate
+        //
+        // Try
+        // v.into_iter().map(|n| (n, f(n))).collect();, from https://www.reddit.com/r/rust/comments/18m24wb/how_to_convert_vec_to_hashmap/
+        // let users_by_name = self.users.into_iter().map(|n| (n.name, n)).collect();
+        // Compilation error = Cannot move
+
+        for user in &mut self.users {
+            let maybe_overlapping_user = _other.users.iter_mut().find(|x| x.name == user.name);
+            let mut balance = user.balance;
+            if let Some(overlapping_user) = maybe_overlapping_user {
+                balance += overlapping_user.balance;
+                overlapping_user.balance = 0;
+            }
+            _other.users.retain(|x| x.name != user.name);
+            merged_users.push(User::new(user.name.clone(), user.credit_line, balance));
+        }
+
+        for non_overlapping_user in &_other.users {
+            merged_users.push(User::new(
+                non_overlapping_user.name.clone(),
+                non_overlapping_user.credit_line,
+                non_overlapping_user.balance,
+            ));
+        }
+
+        Bank {
+            users: merged_users,
+            name: self.name.clone(),
+            credit_interest: self.credit_interest,
+            debit_interest: self.debit_interest,
+        }
+    }
+}
+
+impl Bank {
+    pub(crate) fn accrue_interest(&mut self) {
+        for user in self.users.iter_mut() {
+            let applicable_interest = match user.balance >= 0 {
+                true => self.debit_interest,
+                false => self.credit_interest,
+            };
+            user.balance += user.balance * applicable_interest as i64 / 100;
+        }
+    }
+}
+
+impl Bank {}
+
+#[derive(Debug)]
+struct Balance {
+    value: i64,
+}
```

Comment:
I don't think you use this type anywhere in your code.

Reply:
it is used (although just a wrapper), in e.g.  `fn transfer_funds_happy_path()`

Reply:
This type is used only for tests, so you probably should move it to the tests module. But I think it's better to just remove it, same for `BankHelper`. You can add `balance_of_user` method directly to the `Bank` type and make it return `i64`.